src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -0,0 +1,130 @@
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet takes grocery list and displays it to the screen based on the users family*/
+@WebServlet("/grocery-list")
+public class GroceryServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {      
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    // if user isn't associated with family don't allow them to create a list
+    Query query = new Query("UserInfo")
+        .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
+    PreparedQuery results = datastore.prepare(query);
+    Entity userInfoEntity = results.asSingleEntity();
+    if (userInfoEntity == null) {
+        response.sendRedirect("/grocery.html");
+        return;
+    }
+    String assignGrocery = request.getParameter("assignGrocery");
+    String noneAssigned = " ";
+    Entity groceryEntity = new Entity("Grocery");
+   
+    // turn familyID into a key so the users family can be accessed
+    long familyID = (long) userInfoEntity.getProperty("familyID");
+    Key familyEntityKey = KeyFactory.createKey("Family", familyID);
+    Entity familyEntity;
+    try {
+        familyEntity = datastore.get(familyEntityKey);
+    } catch (EntityNotFoundException e) {
+        System.out.println("Family not found");
+        return;
+    }
+
+    // checks if the given email matches a email in the family
+    ArrayList<String> memberEmails = (ArrayList<String>) familyEntity.getProperty("memberEmails");
+    for(int i = 0; i < memberEmails.size(); i++) {
+        System.out.println("Cuerrnt email is : " + memberEmails.get(i));
+        if(assignGrocery.equals(memberEmails.get(i))) {
+            groceryEntity.setProperty("assignEmail" , memberEmails.get(i));
+            System.out.println("emails match");
+            break;
+        } else {
+            groceryEntity.setProperty("assignEmail", noneAssigned);
+        }
+    }
+    
+    // adds item to datastore
+    String grocery = request.getParameter("groceryItem");
+    if( grocery != null && !grocery.equals("")) {
+        groceryEntity.setProperty("grocery", grocery);
+        groceryEntity.setProperty("familyID", familyID);
+        datastore.put(groceryEntity);
+    }
+               
+    doGet(request,response);  
```

Comment:
I think Jenny mentioned it in a comment on someone's PR, it might be worth de-coupling the `doPost` and `doGet` functionality instead of always doing a `doGet` at the end of the `doPost`. Instead, make the frontend invoke `doPost` first and then `doGet`.